Gruntfile.js

```diff
@@ -362,12 +362,18 @@ module.exports = function(grunt) {
   grunt.registerTask("dist-compile", ["test", "uglify", "compress"]);
 
   grunt.registerTask("commitjs", ["dist-compile", "gitcommit:built"]);
-  grunt.registerTask("default", ["connect", "dev-compile", "watch"]);
+  grunt.registerTask("default", ["connect", "dev-compile", "watch-silent"]);
 
   grunt.registerTask("test", ["dev-compile", "test-local"]);
   grunt.registerTask("test-local", ["blanket_mocha", "ts:verifyDefinitionFiles", "lint"]);
   grunt.registerTask("test-sauce", ["connect", "saucelabs-mocha"]);
 
+  grunt.registerTask("watch-silent", function() {
+    // Surpresses the "Running 'foo' task" messages. More info at #2507.
+    grunt.log.header = function() {};
```

Comment:
Oh so this is not a HACKHACK?  If it's not why is an issue being addressed here?

Also what does `grunt.log.header` do?